src/main/groovy/wooga/gradle/build/unity/UnityBuildPlugin.groovy

```diff
@@ -58,6 +60,44 @@ class UnityBuildPlugin implements Plugin<Project> {
                 conventionMapping.map("buildPlatform", {extension.getDefaultPlatform()})
                 conventionMapping.map("toolsVersion", {extension.getToolsVersion()})
                 conventionMapping.map("outputDirectoryBase", {extension.getOutputDirectoryBase()})
+                conventionMapping.map("inputFiles", {
```

Comment:
So in general you'd say it's better to do input analysis logic outside of tasks and put it into plugins instead, right? Before you had it in task's getInputFiles and moved outside. Looks cleaner to me as well. 

Or is it just because you couldn't do something from a task?

Reply:
It is just a split of concerns. The task here says: "I need these inputFiles to run" Which files that are doesn't matter for the task itself. What the plugin also provides is a convention. It states that by default, we set these inputFiles for you. And this setup is not handled in the task for simple code re-usage. I could decide that this input setting is actually better placed in the parent task type `UnityTask`. Or I could apply the same convention mapping to multiple task types which share an interface.